API.md

```diff
@@ -5030,3 +5030,68 @@ const plugin = {
     }
 };
 ```
+
+## Diagnostic Channels
+
+Diagnostic Channels allows Hapi to report events to other modules. This is useful for APM vendors to collect data from Hapi for diagnostics purposes.
+
+See [the official documentation](https://nodejs.org/docs/latest/api/diagnostics_channel.html) for more information.
+
+### `hapi.onServer`
+
+This event is sent when a new server is created. The [`server`](#server) object is passed as the message.
+
+```js
+const DC = require('diagnostics_channel');
+const channel = DC.channel('hapi.onServer');
+
+channel.subscribe((server) => {
+    // Do something with the server
+    console.log(server.version);
+});
```

Comment:
Hapi itself doesn't actually use any deprecated API, only these examples and the tests. I'm not a fan of using deprecated API to illustrate how to use, but since this is targeted at APM vendors they should known what they are doing anyway.

Maybe just have a single example at the top?

Reply:
I don't mind doing another PR when Hapi drops support for Node 14 to replace those deprecated usage